data-model-translator-wrapper/src/main/java/ai/aitia/arrowhead/dmtpw/service/DataModelTranslationService.java

```diff
@@ -0,0 +1,162 @@
+        return new DataModelTranslationResultResponseDTO(job.getStatus(), result, job.getResultMimeType());
+    }
+
+	//=================================================================================================
+	// assistant methods
+
+    //-------------------------------------------------------------------------------------------------
+    private void doTranslationJob(final DataModelTranslationTask job) {
+    	// start the job
+    	job.setStatus(DataModelTranslationTaskStatus.IN_PROGRESS);
+
+    	// decode the bytes
+    	final byte[] bytes = Base64.getDecoder().decode(job.getPayload());
+
+    	final String fileName = job.getUuid().toString() + ".xml";
+    	try {
+    		// save the file
+    		Files.write(Paths.get(sysInfo.getInputFolder(), fileName), bytes);
+
+    		// run the script
+    		runTranslationScript(job.getUuid().toString());
+    	} catch (final Exception ex) {
+    		job.setStatus(DataModelTranslationTaskStatus.ERROR);
+    		throw new InternalServerError("An error occured while running the translation script");
+    	}
+
+    	fileNameCache.put(job.getUuid(), fileName);
+
+    	// the job is done
+    	job.setStatus(DataModelTranslationTaskStatus.DONE);
+    }
+
+  //-------------------------------------------------------------------------------------------------
+    private void runTranslationScript(final String name) throws ConfigurationException, InterruptedException, IOException {
+    	
+        try {
+            final ProcessBuilder translationScriptProcessBuilder = new ProcessBuilder("python", sysInfo.getTranslationScriptLocation(), name);
+            translationScriptProcessBuilder.inheritIO();
+            translationScriptProcessBuilder.directory(new File(new File(sysInfo.getTranslationScriptLocation()).getParent()));
+            final Process translationScriptProcess = translationScriptProcessBuilder.start();
+            final int exitCode = translationScriptProcess.waitFor();
+            if (exitCode != 0) {
+            	logger.debug("Translation python script exited with code: " + exitCode);
+            }
+        } catch (final Exception ex) {
+			throw ex;
+        }
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    private String readTranslatedFile(final String filename) {
+
+    	if (filename == null) {
+    		return null;
+    	}
+
+        try {
+            final byte[] fileBytes = Files.readAllBytes(Paths.get(sysInfo.getOutputFolder(), filename));
+            return Base64.getEncoder().encodeToString(fileBytes);
+        } catch (final IOException ex) {
+        	logger.error(ex.getMessage());
+            return null;
+        }
+    }
+
+	//-------------------------------------------------------------------------------------------------
+    private void start() {
+        final Thread worker = new Thread(() -> {
+            while (true) {
```

Comment:
instead of true you should use a boolean member and you can set that to false in the customDestroy() of your application init listener.